internal/operands/vm-console-proxy/reconcile.go

```diff
@@ -0,0 +1,238 @@
+
+type vmConsoleProxy struct {
+	serviceAccount     *core.ServiceAccount
+	clusterRole        *rbac.ClusterRole
+	clusterRoleBinding *rbac.ClusterRoleBinding
+	service            *core.Service
+	deployment         *apps.Deployment
+	configMap          *core.ConfigMap
+}
+
+var _ operands.Operand = &vmConsoleProxy{}
+
+func New(bundle *vm_console_proxy_bundle.Bundle) *vmConsoleProxy {
+	return &vmConsoleProxy{
+		serviceAccount:     &bundle.ServiceAccount,
+		clusterRole:        &bundle.ClusterRole,
+		clusterRoleBinding: &bundle.ClusterRoleBinding,
+		service:            &bundle.Service,
+		deployment:         &bundle.Deployment,
+		configMap:          &bundle.ConfigMap,
+	}
+}
+
+func (v *vmConsoleProxy) Name() string {
+	return operandName
+}
+
+func (v *vmConsoleProxy) WatchTypes() []operands.WatchType {
+	return []operands.WatchType{
+		{Object: &core.ServiceAccount{}},
+		{Object: &core.Service{}},
+		{Object: &apps.Deployment{}},
+		{Object: &core.ConfigMap{}},
+	}
+}
+
+func (v *vmConsoleProxy) WatchClusterTypes() []operands.WatchType {
+	return []operands.WatchType{
+		{Object: &rbac.ClusterRole{}},
+		{Object: &rbac.ClusterRoleBinding{}},
+	}
+}
+
+func (v *vmConsoleProxy) RequiredCrds() []string {
+	return nil
+}
+
+func (v *vmConsoleProxy) Reconcile(request *common.Request) ([]common.ReconcileResult, error) {
+	var results []common.ReconcileResult
+	var reconcileFunc []common.ReconcileFunc
+
+	if !isEnabled(request) {
+		v.Cleanup(request)
+		return []common.ReconcileResult{}, nil
+	}
+
+	reconcileFunc = append(reconcileFunc, reconcileServiceAccountsFuncs(*v.serviceAccount))
+	reconcileFunc = append(reconcileFunc, reconcileClusterRoleFuncs(*v.clusterRole))
+	reconcileFunc = append(reconcileFunc, reconcileClusterRoleBindingFuncs(*v.clusterRoleBinding))
+	reconcileFunc = append(reconcileFunc, reconcileConfigMapFuncs(*v.configMap))
+	reconcileFunc = append(reconcileFunc, reconcileServiceFuncs(*v.service))
+	reconcileFunc = append(reconcileFunc, reconcileDeploymentFuncs(*v.deployment))
+
+	reconcileBundleResults, err := common.CollectResourceStatus(request, reconcileFunc...)
+	if err != nil {
+		return nil, err
+	}
+
+	return append(results, reconcileBundleResults...), nil
+}
+
+func (v *vmConsoleProxy) Cleanup(request *common.Request) ([]common.CleanupResult, error) {
+	var objects []client.Object
+
+	objects = append(objects, v.serviceAccount.DeepCopy())
+	objects = append(objects, v.clusterRole.DeepCopy())
+	objects = append(objects, v.clusterRoleBinding.DeepCopy())
+	objects = append(objects, v.configMap.DeepCopy())
+	objects = append(objects, v.service.DeepCopy())
+	objects = append(objects, v.deployment.DeepCopy())
+
+	return common.DeleteAll(request, objects...)
+}
+
+func reconcileServiceAccountsFuncs(serviceAccount core.ServiceAccount) common.ReconcileFunc {
+	return func(request *common.Request) (common.ReconcileResult, error) {
+		namespace := request.Instance.Namespace
+		serviceAccount.Namespace = namespace
```

Comment:
The namespace for all resources here should be `kubevirt`.

Reply:
Okay, I wasn't sure about it. So then just `serviceAccount.Namespace = "kubevirt"`?

Reply:
For now. Then we can fix it in a later PR.